### ISSUE TITLE
Allow disabling retainFocus after selection

### ISSUE THREAD

**Issue by bartsyst** [AbdulRahmanAlHamali/flutter_typeahead#545]
I think it would be really nice to add some param in TypeAheadField or SuggestionsController to modify retainFocus in SuggestionsFieldSelectConnector. It can be "true" by default (like it is right now), as it is in SuggestionsFieldSelectConnector, line 24.

```
void _onSelected(T value) {
  onSelected?.call(value);
  if (hideOnSelect ?? true) {
    controller.close(retainFocus: true);
  } else {
    controller.focusField();
  }
}
```

The reason for this request is that forcing focus retention on hiding interfere with modifying TextEditingController outside TypeAheadField (when declaring own instance of TextEditingController to be used along with TypeAheadField). For example, after suggestion is selected in TypeAheadField, you want to close suggestions box (which occurs automatically with hideOnSelect set to "true"), close keyboard and modify text field with selected data using TextEditingController built-in methods.

`_textEditingController.text = 'selected text'`
or
`_textEdtingController.value = TextEditingValue(text: 'selected text')`

In such case, when retainFocus is "true", listeners inside text field invoke suggestions box and keyboard to be opened.

Furthermore, if you place button on text field using prefix/suffix params, to be used to clean current non-empty value and retainFocus is "true", then tap event on this button also invokes both keyboard and suggestions box to be displayed.

Short video of default behavior for retainFocus (true) and setting value to instance of TextEditingController.

```
return TypeAheadField<Location>(
  onSelected: (location) {
    _textEditingController.value = TextEditingValue(
      text: location.name,
    );
  },
);
```

https://github.com/AbdulRahmanAlHamali/flutter_typeahead/assets/153091522/b9ac1d83-87cb-45f4-986f-c0d5c419fd00

I do not know if it is a bug or not, but using "false" for retainFocus results with both keyboard nor suggestions box to be displayed when setting text for text field or using additional widgets inside text field.

One more short video, with retainFocus set to "false" - tap event on button do not invoke unnecessary keyboard or suggestions box appearance.

https://github.com/AbdulRahmanAlHamali/flutter_typeahead/assets/153091522/284004d5-5ec6-43cf-b905-409bc9fcf4cb

Giving an option to define retainFocus would be nice solution for those who do not want retain focus.

This issue applies to version 5.0.x of TypeAheadField, it was ok in older versions, but I guess it was because there was no usage of TextEditingController which can notify listeners and invoke further events.

**Comment by clragon** [comment]
I have to look into this and see how exactly this is caused and if disabling `retainFocus` is the right solution here.

but as a workaround right now, you can define your own `SuggestionsController` subclass that overrides the `retainFocus` boolean getter to always return false.

**Comment by bartsyst** [comment]
I did just like you mentioned, custom SuggestionsController for workaround, but param would be better, because then we do not need to override package classes. By the way, really nice package :-) 

Try to look at it if you have space, maybe there is something not correct with focus itself. As I was working with version 5.0.1 and in terms of retaining focus set to "true" by default - I also found strange situation that opening menu drawer (so-called hamburger menu with three lines on top left corner) which covers text field with TypeAheadField sometimes invokes keyboard and suggestions box to be shown on screen (when they were previously closed). When retainFocus is set to "false" menu drawer is not invoking keyboard nor suggestions box to be shown.
